src/profile_manager/views.py

```diff
@@ -187,6 +188,16 @@ def get_context_data(self, **kwargs):
         context['badge_assertions_dict_items'] = BadgeAssertion.objects.badge_assertions_dict_items(profile.user)
         context['tags'] = get_user_tags_and_xp(profile.user)
 
+        tags_xp = get_user_tags_and_xp(profile.user)
+
+        # add tags user didn't gain xp in
+        if SiteConfig.get().show_all_tags_on_profiles and tags_xp:
+            tags, _ = zip(*tags_xp)
+            unrelated_tags = Tag.objects.exclude(name__in=tags).order_by('name')
+            tags_xp += [(t, 0) for t in unrelated_tags]
+
+        context['tags'] = tags_xp
+
```

Comment:
your change is the exact same as mine. regardless, if SiteConfig.get().show_all_tags_on_profiles is false then it would not check the condition of tags_xp because of the 'and' statement